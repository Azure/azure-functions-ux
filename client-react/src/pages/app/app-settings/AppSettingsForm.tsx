import * as React from 'react';
import { translate } from 'react-i18next';

import { style } from 'typestyle';
import Stacks from './GeneralSettings/Stacks';
import Platform from './GeneralSettings/Platform';
import { Pivot, PivotItem } from 'office-ui-fabric-react/lib-commonjs/Pivot';
import ApplicationSettings from './ApplicationSettings/ApplicationSettings';
import { AppSettingsFormValues } from './AppSettings.Types';
import { FormikProps } from 'formik';
import ConnectionStrings from './ConnectionStrings/ConnectionStrings';
import DefaultDocuments from './DefaultDocuments/DefaultDocuments';
import HandlerMappings from './HandlerMappings/HandlerMappings';
import VirtualApplications from './VirtualApplications/VirtualApplications';
import Debug from './GeneralSettings/Debugging';
import SlotAutoSwap from './GeneralSettings/SlotAutoSwap';

export const settingsWrapper = style({
  paddingLeft: '15px',
  padding: '5px 20px',
  width: '535px',
});

const defaultDocumentsWrapper = style({
  width: '565px',
});
class AppSettingsForm extends React.Component<FormikProps<AppSettingsFormValues>, any> {
  public render() {
    return (
      <Pivot>
        <PivotItem linkText="General Settings">
          <h3>Stack Settings</h3>
          <div className={settingsWrapper}>
            <Stacks {...this.props} />
          </div>
          <h3>Platform Settings</h3>
          <div className={settingsWrapper}>
            <Platform {...this.props} />
          </div>
          <h3>Debugging</h3>
          <div className={settingsWrapper}>
            <Debug {...this.props} />
          </div>
          <SlotAutoSwap {...this.props} />
        </PivotItem>
        <PivotItem linkText="Application Settings">
          <h3>Application Settings</h3>
          <ApplicationSettings {...this.props} />
          <h3>Connection Strings</h3>
          <ConnectionStrings {...this.props} />
        </PivotItem>
        <PivotItem linkText="Default Documents">
          <h3>Default Documents</h3>
          <div className={defaultDocumentsWrapper}>
            <DefaultDocuments {...this.props} />
          </div>
        </PivotItem>
        <PivotItem linkText="Path Mappings">
          <h3>Handler Mappings</h3>
          <HandlerMappings {...this.props} />
          <h3>Virtual applications and directories</h3>
          <VirtualApplications {...this.props} />
        </PivotItem>
      </Pivot>
    );
  }
}

export default translate('translation')(AppSettingsForm);
