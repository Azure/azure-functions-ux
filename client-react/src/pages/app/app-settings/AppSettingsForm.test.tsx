import { shallow } from 'enzyme';
import * as React from 'react';
import AppSettingsForm from './AppSettingsForm';
const AppSettingsFormAny = AppSettingsForm as any;
describe('App Settings', () => {
  describe('Shallow Testing', () => {
    let container;

    beforeEach(() => {
      container = shallow(<AppSettingsFormAny />);
    });

    it('renders without crashing', () => {
      expect(container.length).toBe(1);
    });
  });
});
