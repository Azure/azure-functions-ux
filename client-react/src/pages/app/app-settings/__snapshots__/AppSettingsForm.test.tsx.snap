// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Settings Shallow Testing render matches snapshot: DefaultSnapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppSettingsForm
    t={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <PivotItem
          linkText="General Settings"
        >
          <h3>
            Stack Settings
          </h3>
          <div
            className="f1bh3whb"
          >
            <Connect(Stacks)
              t={[Function]}
            />
          </div>
          <h3>
            Platform Settings
          </h3>
          <div
            className="f1bh3whb"
          >
            <Platform
              t={[Function]}
            />
          </div>
          <h3>
            Debugging
          </h3>
          <div
            className="f1bh3whb"
          >
            <Debug
              t={[Function]}
            />
          </div>
        </PivotItem>,
        <PivotItem
          linkText="Application Settings"
        >
          <h3>
            Application Settings
          </h3>
          <ApplicationSettings
            t={[Function]}
          />
          <h3>
            Connection Strings
          </h3>
          <ConnectionStrings
            t={[Function]}
          />
        </PivotItem>,
        <PivotItem
          linkText="Default Documents"
        >
          <h3>
            Default Documents
          </h3>
          <div
            className="f1nqrmg7"
          >
            <DefaultDocuments
              t={[Function]}
            />
          </div>
        </PivotItem>,
        <PivotItem
          linkText="Path Mappings"
        >
          <h3>
            Handler Mappings
          </h3>
          <HandlerMappings
            t={[Function]}
          />
          <h3>
            Virtual applications and directories
          </h3>
          <VirtualApplications
            t={[Function]}
          />
        </PivotItem>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <h3>
              Stack Settings
            </h3>,
            <div
              className="f1bh3whb"
            >
              <Connect(Stacks)
                t={[Function]}
              />
            </div>,
            <h3>
              Platform Settings
            </h3>,
            <div
              className="f1bh3whb"
            >
              <Platform
                t={[Function]}
              />
            </div>,
            <h3>
              Debugging
            </h3>,
            <div
              className="f1bh3whb"
            >
              <Debug
                t={[Function]}
              />
            </div>,
          ],
          "linkText": "General Settings",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Stack Settings",
            },
            "ref": null,
            "rendered": "Stack Settings",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Connect(Stacks)
                t={[Function]}
              />,
              "className": "f1bh3whb",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "t": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Platform Settings",
            },
            "ref": null,
            "rendered": "Platform Settings",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Platform
                t={[Function]}
              />,
              "className": "f1bh3whb",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "t": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Debugging",
            },
            "ref": null,
            "rendered": "Debugging",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Debug
                t={[Function]}
              />,
              "className": "f1bh3whb",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "t": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <h3>
              Application Settings
            </h3>,
            <ApplicationSettings
              t={[Function]}
            />,
            <h3>
              Connection Strings
            </h3>,
            <ConnectionStrings
              t={[Function]}
            />,
          ],
          "linkText": "Application Settings",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Application Settings",
            },
            "ref": null,
            "rendered": "Application Settings",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "t": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Connection Strings",
            },
            "ref": null,
            "rendered": "Connection Strings",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "t": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <h3>
              Default Documents
            </h3>,
            <div
              className="f1nqrmg7"
            >
              <DefaultDocuments
                t={[Function]}
              />
            </div>,
          ],
          "linkText": "Default Documents",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Default Documents",
            },
            "ref": null,
            "rendered": "Default Documents",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <DefaultDocuments
                t={[Function]}
              />,
              "className": "f1nqrmg7",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "t": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <h3>
              Handler Mappings
            </h3>,
            <HandlerMappings
              t={[Function]}
            />,
            <h3>
              Virtual applications and directories
            </h3>,
            <VirtualApplications
              t={[Function]}
            />,
          ],
          "linkText": "Path Mappings",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Handler Mappings",
            },
            "ref": null,
            "rendered": "Handler Mappings",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "t": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Virtual applications and directories",
            },
            "ref": null,
            "rendered": "Virtual applications and directories",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "t": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <PivotItem
            linkText="General Settings"
          >
            <h3>
              Stack Settings
            </h3>
            <div
              className="f1bh3whb"
            >
              <Connect(Stacks)
                t={[Function]}
              />
            </div>
            <h3>
              Platform Settings
            </h3>
            <div
              className="f1bh3whb"
            >
              <Platform
                t={[Function]}
              />
            </div>
            <h3>
              Debugging
            </h3>
            <div
              className="f1bh3whb"
            >
              <Debug
                t={[Function]}
              />
            </div>
          </PivotItem>,
          <PivotItem
            linkText="Application Settings"
          >
            <h3>
              Application Settings
            </h3>
            <ApplicationSettings
              t={[Function]}
            />
            <h3>
              Connection Strings
            </h3>
            <ConnectionStrings
              t={[Function]}
            />
          </PivotItem>,
          <PivotItem
            linkText="Default Documents"
          >
            <h3>
              Default Documents
            </h3>
            <div
              className="f1nqrmg7"
            >
              <DefaultDocuments
                t={[Function]}
              />
            </div>
          </PivotItem>,
          <PivotItem
            linkText="Path Mappings"
          >
            <h3>
              Handler Mappings
            </h3>
            <HandlerMappings
              t={[Function]}
            />
            <h3>
              Virtual applications and directories
            </h3>
            <VirtualApplications
              t={[Function]}
            />
          </PivotItem>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <h3>
                Stack Settings
              </h3>,
              <div
                className="f1bh3whb"
              >
                <Connect(Stacks)
                  t={[Function]}
                />
              </div>,
              <h3>
                Platform Settings
              </h3>,
              <div
                className="f1bh3whb"
              >
                <Platform
                  t={[Function]}
                />
              </div>,
              <h3>
                Debugging
              </h3>,
              <div
                className="f1bh3whb"
              >
                <Debug
                  t={[Function]}
                />
              </div>,
            ],
            "linkText": "General Settings",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Stack Settings",
              },
              "ref": null,
              "rendered": "Stack Settings",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(Stacks)
                  t={[Function]}
                />,
                "className": "f1bh3whb",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "t": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Platform Settings",
              },
              "ref": null,
              "rendered": "Platform Settings",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Platform
                  t={[Function]}
                />,
                "className": "f1bh3whb",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "t": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Debugging",
              },
              "ref": null,
              "rendered": "Debugging",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Debug
                  t={[Function]}
                />,
                "className": "f1bh3whb",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "t": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <h3>
                Application Settings
              </h3>,
              <ApplicationSettings
                t={[Function]}
              />,
              <h3>
                Connection Strings
              </h3>,
              <ConnectionStrings
                t={[Function]}
              />,
            ],
            "linkText": "Application Settings",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Application Settings",
              },
              "ref": null,
              "rendered": "Application Settings",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "t": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Connection Strings",
              },
              "ref": null,
              "rendered": "Connection Strings",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "t": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <h3>
                Default Documents
              </h3>,
              <div
                className="f1nqrmg7"
              >
                <DefaultDocuments
                  t={[Function]}
                />
              </div>,
            ],
            "linkText": "Default Documents",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Default Documents",
              },
              "ref": null,
              "rendered": "Default Documents",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <DefaultDocuments
                  t={[Function]}
                />,
                "className": "f1nqrmg7",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "t": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <h3>
                Handler Mappings
              </h3>,
              <HandlerMappings
                t={[Function]}
              />,
              <h3>
                Virtual applications and directories
              </h3>,
              <VirtualApplications
                t={[Function]}
              />,
            ],
            "linkText": "Path Mappings",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Handler Mappings",
              },
              "ref": null,
              "rendered": "Handler Mappings",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "t": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Virtual applications and directories",
              },
              "ref": null,
              "rendered": "Virtual applications and directories",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "t": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
